File: CGtk/CGtk/Article.cs
using System;
namespace CGtk {
    public class Article {
        public Article() {
        }

        public ulong Id { get; set; }
        public string Name { get; set; }
        public Categoria Category { get; set; }
        public float Price { get; set; }

        public Article(ulong id, string name, Categoria categoria, float price) {
            Id = id;
            Name = name;
            Category = categoria;
            Price = price;
        }

    }
}

File: GHibernate/src/main/java/model/Article.java
package model;

import java.math.BigDecimal;

import javax.persistence.*;

@Entity
@Table(name = "articulo")
public class Article {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	@Column(name="nombre")
	private String name;
	@Column(name="precio")
	private BigDecimal price;
	@Column(name="categoria")
	@ManyToOne
	@JoinColumn(name = "categoria")
	private Category category;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public BigDecimal getPrice() {
		return price;
	}

	public void setPrice(BigDecimal price) {
		this.price = price;
	}

	public Category getCategory() {
		return category;
	}

	public void setCategory(Category category) {
		this.category = category;
	}

	@Override
	public String toString() {
		return name + "(" + price.toString() + ")";
	}
}
